spark-cassandra-connector/src/main/scala/com/datastax/spark/connector/types/TupleType.scala

```diff
@@ -34,38 +36,53 @@ case class TupleType(componentTypes: TupleFieldDef*)
 
   override def columns = componentTypes.toIndexedSeq
 
-  override def scalaTypeTag = TupleValue.TupleValueTypeTag
+  override def scalaTypeTag = TupleValue.TypeTag
 
   override def isCollection = false
 
+  private val defaultComponentConverters =
+    componentTypes.map(_.columnType.converterToCassandra).toIndexedSeq
+
   /** Creates new tuple from components converted each to the
     * type determined by an appropriate componentType.
     * Throws IllegalArgumentException if the number of components does
     * not match the number of components in the tuple type. */
-  def newTuple(componentValues: Any*): TupleValue = {
+  def newInstance
+      (componentConverters: IndexedSeq[TypeConverter[_ <: AnyRef]])
+      (componentValues: Any*): TupleValue = {
     require(
       componentValues.length == columns.length,
       s"Expected ${columns.length} components, instead of ${componentValues.length}")
     val values =
       for (i <- columns.indices) yield
-        columnTypes(i).converterToCassandra.convert(componentValues(i))
+        componentConverters(i).convert(componentValues(i))
     new TupleValue(values: _*)
```

Comment:
You can't create a TupleValue *with* conversion, without knowing the types of the columns in the tuple. That's why TupleType serves as a factory here.